pkg/controller/direct/cloudbuild/workerpool_controller.go

```diff
@@ -98,10 +97,34 @@ func (m *model) AdapterForObject(ctx context.Context, reader client.Reader, u *u
 	if projectID == "" {
 		return nil, fmt.Errorf("cannot resolve project")
 	}
-
 	// Get location
 	location := obj.Spec.Location
 
+	var id *CloudBuildWorkerPoolIDentity
+
+	externalRef := direct.ValueOf(obj.Status.ExternalRef)
+	if externalRef == "" {
+		id = fromRaw(projectID, location, resourceID)
+	} else {
+		id, err = asID(externalRef)
+		if err != nil {
+			return nil, err
+		}
+		if id.project != projectID {
```

Comment:
I agree this is a good idea.

I wonder if we can just struct comparison (maybe reflect.DeepEqual ... we just want to make sure that non-exported fields are compared)